File: android/app/src/main/kotlin/com/sopro/sopro/logging/LoggerConfiguration.kt
package com.sopro.sopro.logging

import com.sopro.sopro.BuildConfig

// Configuração central do sistema de logging Kotlin — espelho do LoggerConfiguration Flutter.
// Todos os campos mutáveis para permitir ajuste em runtime.
object LoggerConfiguration {

    // Versão do schema JSON. Incrementar apenas em mudanças incompatíveis de formato.
    const val SCHEMA_VERSION: Int = 1

    // Ativa TRACE/DEBUG e desabilita mascaramento de campos de diagnóstico.
    // Padrão: BuildConfig.DEBUG (true em debug, false em release).
    var debugLogging: Boolean = BuildConfig.DEBUG

    // Nível mínimo fora do modo debug. Em debug, nível efetivo é sempre TRACE.
    var minimumLevel: LogLevel = LogLevel.INFO

    // Habilita envio ao Supabase (fase futura — reservado para AppLogger Kotlin).
    var enableSupabase: Boolean = true

    // Habilita saída no Logcat via android.util.Log.
    var enableConsole: Boolean = BuildConfig.DEBUG

    // Inclui JSON formatado na linha do Logcat (verbose, somente em debug).
    var enablePrettyPrint: Boolean = BuildConfig.DEBUG

    // Aplica LogSanitizer sobre o payload antes de qualquer saída.
    var enableDataMasking: Boolean = true

    // ── Fase futura: fila e retry ─────────────────────────────────────────────
    var maxQueueSize: Int = 100
    var batchSize: Int = 10
    var retryAttempts: Int = 3

    // ── Metadados do app ─────────────────────────────────────────────────────
    var appVersion: String = "0.1.0"
    var buildNumber: String = "1"
    var platform: String = "android"
}

File: lib/infrastructure/logging/core/logger_configuration.dart
import 'package:flutter/foundation.dart';

// Configuração central do sistema de logging.
// Todos os parâmetros são estáticos e mutáveis, permitindo ajuste em runtime
// (ex.: habilitar verbose logging em QA sem rebuild).
class LoggerConfiguration {
  LoggerConfiguration._();

  // Ativa trace/debug e desabilita mascaramento de campos de diagnóstico.
  // Em produção deve ser false.
  static bool debugLogging = kDebugMode;

  // Habilita envio de eventos para o Supabase.
  static bool enableSupabase = true;

  // Habilita saída de log no console (debugPrint).
  static bool enableConsole = kDebugMode;

  // Formata a saída do console com JSON indentado em vez de uma linha só.
  static bool enablePrettyPrint = kDebugMode;

  // Aplica LogSanitizer sobre o payload antes de persistir/enviar.
  static bool enableDataMasking = true;

  // Tamanho máximo da fila de eventos pendentes (fase futura — queue).
  static int maxQueueSize = 100;

  // Número de eventos enviados por lote ao Supabase (fase futura — batch).
  static int batchSize = 10;

  // Tentativas de reenvio em caso de falha de rede (fase futura — retry).
  static int retryAttempts = 3;

  // Versão do app — alimentada pelo AppInitializer na inicialização.
  static String appVersion = '0.1.0';

  // Build number — alimentado pelo AppInitializer na inicialização.
  static String buildNumber = '1';

  // Plataforma alvo (ex.: 'android', 'ios').
  static String platform = 'android';
}
